src/server/pps/server/metrics_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build k8s
+
+package server_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/minikubetestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testutil"
+)
+
+func TestContainerMetrics(t *testing.T) {
+	ctx, cancel := pctx.WithCancel(pctx.TestContext(t))
+	defer cancel()
+	t.Parallel()
+	c, namespace := minikubetestenv.AcquireCluster(t, minikubetestenv.WithPrometheus)
+
+	docsDir := t.TempDir()
+	tmpDir := t.TempDir()
+	cmd := testutil.PachctlBashCmdCtx(ctx, t, c, `
+# run a pipeline in order to have a container to scrape metrics from
+pachctl create project "video-to-frame-traces"
+pachctl create repo raw_videos_and_images
+pachctl put file raw_videos_and_images@master:robot.png -f https://raw.githubusercontent.com/pachyderm/docs-content/main/images/opencv/robot.jpg
+pachctl create pipeline -f - <<EOF
+pipeline:
+  name: video_mp4_converter
+input:
+  pfs:
+    repo: raw_videos_and_images
+    glob: "/*"
+transform:
+  image: lbliii/video_mp4_converter:1.0.14
+  cmd:
+    - python3
+    - /video_mp4_converter.py
+    - --input
+    - /pfs/raw_videos_and_images/
+    - --output
+    - /pfs/out/
+EOF
+# install the podmonitor we document for users
```

Comment:
It is probably dependency hell to get the Go object for this thing, but k8s/client-go has a dynamic.Interface that has a similar API to kubernetes.Interface and would let you apply this from Go instead of Bash.  We use it in starpach; https://github.com/pachyderm/pachyderm/blob/master/src/internal/starlark/lib/k8s/k8s.go#L400